component/core/src/main/java/org/exoplatform/social/core/jpa/storage/entity/SpaceEntity.java

```diff
@@ -177,138 +190,18 @@ public class SpaceEntity implements Serializable {
-    return id;
-  }
-
-  public void setId(Long id) {
-    this.id = id;
-  }
-
-  public String getPrettyName() {
-    return prettyName;
-  }
-
-  public void setPrettyName(String prettyName) {
-    this.prettyName = prettyName;
-  }
-
-  public String getDisplayName() {
-    return displayName;
-  }
-
-  public void setDisplayName(String displayName) {
-    this.displayName = displayName;
-  }
-
-  public SpaceRegistration getRegistration() {
-    return registration;
-  }
-
-  public void setRegistration(SpaceRegistration registration) {
-    this.registration = registration;
-  }
-
-  public String getDescription() {
-    return description;
-  }
-
-  public void setDescription(String description) {
-    this.description = description;
-  }
-
-  public Date getAvatarLastUpdated() {
-    return avatarLastUpdated;
-  }
-
-  public void setAvatarLastUpdated(Date avatarLastUpdated) {
-    this.avatarLastUpdated = avatarLastUpdated;
-  }
-
-  public Date getBannerLastUpdated() {
-    return bannerLastUpdated;
-  }
-
-  public void setBannerLastUpdated(Date bannerLastUpdated) {
-    this.bannerLastUpdated = bannerLastUpdated;
-  }
-
-  public SpaceVisibility getVisibility() {
-    return visibility;
-  }
-
-  public void setVisibility(SpaceVisibility visibility) {
-    this.visibility = visibility;
-  }
-
-  public String getGroupId() {
-    return groupId;
-  }
-
-  public void setGroupId(String groupId) {
-    this.groupId = groupId;
-  }
-
-  public String getUrl() {
-    return url;
-  }
-
-  public void setUrl(String url) {
-    this.url = url;
-  }
-
-  public Date getCreatedDate() {
-    return createdDate;
-  }
-
-  public void setCreatedDate(Date createdDate) {
-    this.createdDate = createdDate;
-  }
-
-  public Date getUpdatedDate() {
-    return updatedDate;
-  }
-
-  public void setUpdatedDate(Date updatedDate) {
-    this.updatedDate = updatedDate;
-  }
-
-  public Set<SpaceMemberEntity> getMembers() {
-    return members;
-  }
-
-  public void setMembers(Set<SpaceMemberEntity> members) {
-    this.members = members;
-  }
-
-  public Set<GroupSpaceBindingEntity> getSpaceBindingEntities() {
-    return spaceBindingEntities;
-  }
-
-  public void setSpaceBindingEntities(Set<GroupSpaceBindingEntity> spaceBindingEntities) {
-    this.spaceBindingEntities = spaceBindingEntities;
-  }
-
-  public Set<GroupSpaceBindingReportActionEntity> getSpaceBindingReportEntities() {
-    return spaceBindingReportEntities;
-  }
-
-  public void setSpaceBindingReportEntities(Set<GroupSpaceBindingReportActionEntity> spaceBindingReportEntities) {
-    this.spaceBindingReportEntities = spaceBindingReportEntities;
-  }
-
-  public List<SpaceCategoryEntity> getCategories() {
-    return categories;
-  }
-
-  public void setCategories(List<SpaceCategoryEntity> categories) {
-    this.categories = categories;
-  }
+  @Setter
+  @Getter
+  @ManyToOne
+  @JoinColumn(name = "PARENT_SPACE_ID")
+  private SpaceEntity                              parentSpaceEntity;
```

Comment:
Do we need to map all the Parent Entity ? IMO, we can map it as an id only, since we don't use it in a JPQL. WDYT ?

```suggestion
  private long                              parentSpaceId;
```